### ISSUE TITLE
Cannot set params for yaml component in environment

### ISSUE THREAD

**Issue by cspargo** [ksonnet/ksonnet#575]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
>

**What happened**:

I can set a param for a yaml component
```
$ ks param set nginx metadata.labels '{"test":"value"}'
```

but if I try to set that in an environment, it fails

```
$ ks param set --env minikube nginx metadata.labels '{"test":"value"}'
INFO setting environment component                 component-name=nginx
ERROR set params: invalid field name "metadata.labels"
```

**What you expected to happen**:

should be able to set parameters at the environment level

**How to reproduce it (as minimally and precisely as possible)**:

import a yaml component with `ks import -f component.yaml` and set param (see above)

**Environment**:
- ksonnet version (use `ks version`): 0.10.2
- Kubernetes version (use `kubectl version`): 1.10.2

**Comment by bryanl** [comment]
I'm circling back around this to determine the cause.

**Comment by bryanl** [comment]
I've reviewed this and we fixed this issue in the 0.11 release. Please re-open if you encounter any issues.
